lab2/syntax.in

```diff
@@ -0,0 +1,40 @@
+	<program> ::= START <statementList> END
+	<statementList> ::= <statement> | <statement><statementList>
+	<statement> ::= <simpleStatement> | <structStatement>	
+	<simpleStatement> ::= <defineStatement> | <ioStatement> | <assignStatement>
+	<defineStatement> ::= define <type> <identifier>;
+	<type> ::= <simpleType> | <arrayType>
+	<simpleType> ::= integer | string | bool
+	<arrayType> ::= <simpleType>[<unsignedNumber>]
+	<identifier> ::= <lowercaseLetter> | <identifier><letter> | <identifier><digit>
+	<ioStatement> ::= <readStatement> | <writeStatement>;
+	<readStatement> ::= read(<identifier>)
+	<writeStatement> ::= display(<displayable>)
+	<displayable> ::= <identifier> | <INTEGER> | <STRING> | <BOOL>
+	<assignStatement> ::= <identifier> = <expression>
+	<expression> ::= <term> | <expression> + <term> | <expression> - <term>
+	<term> ::= <factor> | <term> * <factor> | <term> / <factor>
+	<factor> ::= (<expression>) | <identifier> | <INTEGER>
```

Comment:
could the factor also be an item from an array? How do you access the elements of an array?

Reply:
that's right. `<factor>` can also be `<identifier>[<unsignedNumber>]` where <identifier> is the variable that holds an array